### PR TITLE
Add type hints for klein.test.test_resource.

### PR DESCRIPTION
**Pull request by wsanchez**

### CODE REVIEW

File: src/klein/test/test_resource.py [hunk 1]
```diff
@@ -100,8 +101,8 @@ def write(data):
                 "Request.finish was called."
             )
 
-    def getWrittenData():
-        return request._written.getvalue()
+    def getWrittenData() -> bytes:
+        return cast(BytesIO, request._written).getvalue()
```
Comment (wsanchez) [inline src/klein/test/test_resource.py:105]: I think the issue here is that `_written` is sloppily hung off of `request` here so mypy doesn't know it's type.

#476 changes this function to a class, which mypy likes better.

**Review by adiroiban — APPROVED** [review]
Thanks for the update. It looks good.

I did a quick review and I don't see any issues... but I am not a mypy / typing expert.

I will put back @twisted/twisted-contributors back for review to see if someone else has time to check this.

otherwise, I don't think that this should be a blocker.

File: mypy.ini [hunk 2]
```diff
@@ -43,9 +43,6 @@ allow_untyped_defs = True
 [mypy-klein.test.test_plating]
 allow_untyped_defs = True
 
-[mypy-klein.test.test_resource]
-allow_untyped_defs = True
```
Comment (glyph) [inline mypy.ini:47]: Love to see it.

File: src/klein/_interfaces.py [hunk 3]
```diff
@@ -7,18 +7,21 @@
 works, since mypy doesn't otherwise get along with Zope Interface.
 """
 
-from typing import Mapping, Optional
+from typing import Mapping, Optional, Union
 
 from zope.interface import Attribute, Interface
 
 
+KleinQueryValue = Union[str, int, float]
```
Comment (glyph) [inline src/klein/_interfaces.py:15]: I know this is just moving, but seeing it here illustrates to me that we really ought to have put in a little textual description, maybe a `@var`, explaining what it's actually for...

File: src/klein/test/test_resource.py (hunk 1 above)
Comment (glyph) [reply src/klein/test/test_resource.py:105]: Yeah, this kind of ad-hoc nonsense is something that I'm glad Mypy is forcing us to stop doing.

File: src/klein/test/test_resource.py [hunk 4]
```diff
@@ -115,16 +116,18 @@ def getWrittenData():
     return request
 
 
-def _render(resource, request, notifyFinish=True):
+def _render(
+    resource: KleinResource, request: IRequest, notifyFinish: bool = True
+) -> Deferred:
```
Comment (glyph) [inline src/klein/test/test_resource.py:121]: Can we depend on `Deferred` being generic here, and give it a type?

**Review by glyph — APPROVED** [review]
Seems pretty straightforward. Let's go!

File: src/klein/test/test_resource.py (hunk 4 above)
Comment (wsanchez) [reply src/klein/test/test_resource.py:121]: Needs #476

**Pull request merged**